resources/elasticbeanstalk_configuration_settings.go

```diff
@@ -0,0 +1,173 @@
+	return &schema.Table{
+		Name:         "aws_elasticbeanstalk_configuration_settings",
+		Description:  "Describes the settings for a configuration set.",
+		Resolver:     fetchElasticbeanstalkConfigurationSettings,
+		Multiplex:    client.AccountRegionMultiplex,
+		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter: client.DeleteAccountRegionFilter,
+		Options:      schema.TableCreationOptions{PrimaryKeys: []string{"application_name", "date_created"}},
+		Columns: []schema.Column{
+			{
+				Name:        "account_id",
+				Description: "The AWS Account ID of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSAccount,
+			},
+			{
+				Name:        "region",
+				Description: "The AWS Region of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSRegion,
+			},
+			{
+				Name:        "application_name",
+				Description: "The name of the application associated with this configuration set.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "date_created",
+				Description: "The date (in UTC time) when this configuration set was created.",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "date_updated",
+				Description: "The date (in UTC time) when this configuration set was last modified.",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "deployment_status",
+				Description: "If this configuration set is associated with an environment, the DeploymentStatus parameter indicates the deployment status of this configuration set:  * null: This configuration is not associated with a running environment.  * pending: This is a draft configuration that is not deployed to the associated environment but is in the process of deploying.  * deployed: This is the configuration that is currently deployed to the associated running environment.  * failed: This is a draft configuration that failed to successfully deploy.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "description",
+				Description: "Describes this configuration set.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "environment_name",
+				Description: "If not null, the name of the environment for this configuration set.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "platform_arn",
+				Description: "The ARN of the platform version.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "solution_stack_name",
+				Description: "The name of the solution stack this configuration set uses.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "template_name",
+				Description: "If not null, the name of the configuration template for this configuration set.",
+				Type:        schema.TypeString,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:        "aws_elasticbeanstalk_configuration_setting_option_settings",
```

Comment:
nit: maybe `aws_elasticbeanstalk_configuration_setting_option` (without `_settings` at the end) would be a better name